src/Microsoft.IdentityModel.Tokens/SignatureProvider.cs

```diff
@@ -68,6 +68,8 @@ public void Dispose()
         /// <param name="disposing">true, if called from Dispose(), false, if invoked inside a finalizer</param>
         protected abstract void Dispose(bool disposing);
 
+        internal bool IsCached { get; set; }
```

Comment:
Is this property needed? Can we use `SignatureProvider.CryptoProviderCache` for the same purpose? We set `IsCached` and `CryptoProviderCache` around the same place where we add the provider to the cache and we check these properties at the same place in the factory also.

Reply:
@pmaytak you are correct, the only reason i put this is for insurance. If we have a bug somewhere and the CryptoCacheProvider is somehow null then we can dispose.